packages/core/src/utils/graphFacade.ts

```diff
@@ -0,0 +1,66 @@
+import { IGraphite } from '@stoplight/graphite';
+import {
+  createFileSystemBackend,
+  FileSystemBackend,
+  FilesystemNodeType,
+} from '@stoplight/graphite/backends/filesystem';
+import { NodeCategory } from '@stoplight/graphite/graph/nodes';
+import { createGraphite } from '@stoplight/graphite/graphite';
+import { createOas2HttpPlugin } from '@stoplight/graphite/plugins/http/oas2';
+import { createOas3HttpPlugin } from '@stoplight/graphite/plugins/http/oas3';
+import { createJsonPlugin } from '@stoplight/graphite/plugins/json';
+import { createOas2Plugin } from '@stoplight/graphite/plugins/oas2';
+import { IHttpOperation } from '@stoplight/types';
+import * as fs from 'fs';
+import { extname, join } from 'path';
+
+import compact = require('lodash/compact');
+
+export class GraphFacade {
+  private fsBackend: FileSystemBackend;
+  private graphite: IGraphite;
+  private cwd: string;
+
+  constructor() {
+    const graphite = (this.graphite = createGraphite());
+    graphite.registerPlugins(
+      createJsonPlugin(),
+      createOas2Plugin(),
+      createOas2HttpPlugin(),
+      createOas3HttpPlugin()
+    );
+    this.cwd = process.cwd();
+    this.fsBackend = createFileSystemBackend(this.cwd, graphite, fs);
+  }
+
+  public async createFilesystemNode(fsPath: string | undefined) {
+    if (fsPath) {
+      const resourceFile = join(process.cwd(), fsPath);
+      const subtype = extname(resourceFile).slice(1);
+      const stat = fs.lstatSync(resourceFile);
+      if (stat.isDirectory()) {
+        this.graphite.graph.addNode({
+          category: NodeCategory.Source,
+          type: FilesystemNodeType.Directory,
+          path: fsPath,
+        });
+        this.fsBackend.readdir(fsPath);
+      } else if (stat.isFile()) {
+        this.graphite.graph.addNode({
+          category: NodeCategory.Source,
+          type: FilesystemNodeType.File,
+          subtype,
+          path: fsPath,
+        });
+        this.fsBackend.readFile(fsPath);
+      }
+      return await this.graphite.scheduler.drain();
+    }
+    return null;
+  }
+
+  get httpOperations(): IHttpOperation[] {
+    const nodes = this.graphite.graph.virtualNodes.filter(node => node.type === 'http-operation');
+    return compact(nodes.map<IHttpOperation>(node => node.data as IHttpOperation));
```

Comment:
What does it mean if node.data is empty? There was an error on graphite side? The graph hasn't been processed yet? Maybe we should throw here?

Reply:
Good question, not sure what to be honest. Looks like it's in the interface for no particular reason:

```
export interface IVirtualNode extends IBaseNode {
    category: NodeCategory.Virtual;
    data?: unknown;
}
```

In real-life this should not be undefined, though.

Reply:
I'd throw error here since it shouldn't happen. If we keep silent here then we won't even notice that sth is going wrong or it will be very hard to find. If we do throw (keep in mind we are at early stage, things can happen) then we will be able to quickly identify the problem and add compact or resolve bug in graph and change the interface.

But it's just a non-blocking opinion ;)

Reply:
I will make a not to make that `data` field required instead of throwing error. There is no point in having virtual node w/o data imo.

Reply:
It might have sense to data be empty from graph perspective (sth might not yet been processed). It does not from prism perspective (we did scheduler.drain() so we excpect everything to be resolved). That means that it is runtime exception.

Reply:
I think we can probably go with required data prop on virtual nodes - should be set to something when the node is created.